third_party_libraries/kml.js

```diff
@@ -660,6 +648,62 @@ L.Util.extend(L.KML, {
 		return coords;
 	},
 
+
+	/**
+	 * Gets the style options for the placemarks leaflet marker
+	 *
+	 * @param {xml Element} place A placemark XML element
+	 * @param {path} style Style options for the placemark
+	 * @return {path[]} options The style options for the placemark
+	 */
+	getOptions: function (place, style) {
+		var i, j, el, options = {};
+		el = place.getElementsByTagName('styleUrl');
+
+		for (i = 0; i < el.length; i++) {
+			var url = el[i].childNodes[0].nodeValue;
+			for (var a in style[url])
+			{
+				options[a] = style[url][a];
+			}
+		}
+
+		return options;
+	},
+
+	/**
+	 * Creates a base layergroup element for the placemarks leaflet icon
+	 *
+	 * @param {xml Element} place A placemark XML element
+	 * @param {path[]} options The style options for the placemark
+	 * @param {xml} xml Xml data related to the placemark
+	 * @return {FeatureGroup} layer Basic location and style data pertaining to
+	 * 		the placemark
+	 */
+	initializeLayer: function (place, options, xml) {
+		var layers = [];
+		var parse = ['LineString', 'Polygon', 'Point'];
+		for (j in parse) {
+			var tag = parse[j];
```

Comment:
> The for...in statement iterates over the enumerable properties of an object, in arbitrary order. For each distinct property, statements can be executed. ([source](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in))

In the case of an array, that would be it's indices.  `parse[j]` is used to retrieve the value.